spec/services/email_alerts_updater_spec.rb

```diff
@@ -5,55 +5,95 @@
+      let(:expected_unsubscribe_email_message) do
+        <<~BODY
+          This topic has been archived. You will not get any more emails about it.
+
+          You can find more information about this topic at [#{Plek.website_root}#{successor_base_path}](#{Plek.website_root}#{successor_base_path}).
+        BODY
+      end
 
-      expect {
-        described_class.call(
-          item: topic,
-          successor:,
+      before do
+        allow(successor).to receive(:base_path).and_return(successor_base_path)
+        allow(successor).to receive(:mapped_specialist_topic_content_id).and_return("e3c4f270-044b-11ee-be56-0242ac120002")
+        allow(SecureRandom).to receive(:uuid).and_return("some-uuid")
+      end
+
+      it "makes a call the email-alert-api bulk unsubscribe endpoint with the slug, and generated email body and sender_message_id" do
+        stub_email_alert_api_bulk_unsubscribe_with_message(
+          slug: "tax-credits-and-child-benefit-child-benefit",
+          body: expected_unsubscribe_email_message,
+          sender_message_id: "some-uuid",
         )
-      }.to_not raise_error
+
+        expect {
+          described_class.call(
+            item: topic,
+            successor:,
+          )
+        }.to_not raise_error
+      end
+
+      it "raises an error when there is an error from the email_alert_api" do
+        stub_email_alert_api_does_not_have_subscriber_list({ "links" => { topics: %w[ABC] } })
+
+        expect {
+          described_class.call(
+            item: OpenStruct.new(subscriber_list_search_attributes: { "links" => { topics: %w[ABC] } }, content_id: "ABC"),
+            successor:,
+          )
+        }.to raise_error(GdsApi::HTTPNotFound)
+      end
     end
 
-    it "raises an error when there is an error from the email_alert_api" do
-      stub_email_alert_api_does_not_have_subscriber_list({ "links" => { topics: %w[ABC] } })
+    context "when the successor has the same mapped specialist topic id as the topic it is replacing" do
+      let(:successor_data) do
+        {
+          "title" => "Successor document collection",
+          "base_path" => successor_base_path,
+          "content_id" => "835af2ae-ed12-49f0-9b3c-d6795d028484",
+          "document_type" => "document_collection",
+          "description" => "description for Successor document collection",
+          "details" => {
+            "mapped_specialist_topic_content_id" => topic.content_id,
+          },
+        }
+      end
+      let(:successor) { ContentItem.new(successor_data) }
 
-      expect {
-        described_class.call(
-          item: OpenStruct.new(subscriber_list_search_attributes: { "links" => { topics: %w[ABC] } }, content_id: "ABC"),
-          successor:,
+      before do
+        stub_email_alert_api_creates_subscriber_list(
+          "title" => "Successor document collection",
+          "links" => { "document_collections" => [topic.content_id] },
```

Comment:
I know it doesn't matter, but I'd stick a random uiid in here instead, as it's a bit confusing seeing `topic.content_id` which would be used for the `details[mapped_specialist_topic_content_id]`

Reply:
Oh I think that's the point here, i.e. it should be the same thing because it's a mapped specialist topic content id?

Reply:
No, this is the reverse link, so it would be the content id of the document collection page itself. Sorry - complicated! Hope that makes sense 😅